### ISSUE TITLE
Can't log into TEAM

### ISSUE THREAD

**Issue by cemendes** [aws-samples/iam-identity-center-team#150]
### Discussed in https://github.com/aws-samples/iam-identity-center-team/discussions/138

<div type='discussions-op-text'>

<sup>Originally posted by **cemendes** January  6, 2024</sup>
I did a new install of TEAM in a child account (Team Account). All went well, including the execution of the cognito.sh script(no errors) and now after the login, I see the TEAM page loading non-stop. 
![2024-01-06_23-00-27 (1)](https://github.com/aws-samples/iam-identity-center-team/assets/16995556/72f2edf7-cbf5-4d02-a0e6-e1fe1e87909b)

In the team account, I checked cloud trail and I see this error when trying to log-in. 

Any ideas would be appreciated. 

Field | Value
-- | --
1704602118841
730335522437:/aws/lambda/teamgetGroups-main
2024/01/07/[$LATEST]b44cc22778ae429eb5c11da269e60a62
**[ERROR]** ParamValidationError: Parameter validation failed:Invalid type for parameter MemberId.UserId, value: None, type: <class 'NoneType'>, valid types: <class 'str'>Traceback (most recent call last):  File "/var/task/index.py", line 135, in handler    groupData = list_idc_group_membership(userId)  File "/var/task/index.py", line 116, in list_idc_group_membership    for page in paginator:  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 269, in __iter__    response = self._make_request(current_kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/paginate.py", line 357, in _make_request    return self._method(**current_kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call    return self._make_api_call(operation_name, kwargs)  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 919, in _make_api_call    request_dict = self._convert_to_request_dict(  File "/opt/python/lib/python3.9/site-packages/botocore/client.py", line 990, in _convert_to_request_dict    request_dict = self._serializer.serialize_to_request(  File "/opt/python/lib/python3.9/site-packages/botocore/validate.py", line 381, in serialize_to_request    raise ParamValidationError(report=report.generate_report())
1704602111105

</div>

**Comment by cemendes** [comment]
After a troubleshoot session with @tawoyinfa we saw that the the filter UserName has been deprecated in the[ IDC api. ](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/identitystore/client/list_users.html)

Whenever this code is run

```
import json
import boto3

def get_user():
    try:
        client = boto3.client('identitystore')
        response = client.list_users(
            IdentityStoreId='d-91671147cf',
            Filters=[
                {
                    **'AttributePath': 'UserName',**
                    'AttributeValue': 'team-admin'
                },
            ]
        )
        return response['Users'][0]['UserId']
    except ClientError as e:
        print(e.response['Error']['Message'])

def lambda_handler(event, context):
    # TODO implement
    user=get_user()
    print(user)
    return {
        'statusCode': 200,
        'body': json.dumps('Hello from Lambda!')
    }

```
An error appears saying the filter is deprecated.
```

`Test Event Name
(unsaved) test event

Response
{
  "statusCode": 200,
  "body": "\"Hello from Lambda!\""
}

Function Logs
START RequestId: 2636ef37-eaa4-4459-b4da-00222be95df3 Version: $LATEST
**Filter is deprecated, please remove filter from request.**
None
END RequestId: 2636ef37-eaa4-4459-b4da-00222be95df3
REPORT RequestId: 2636ef37-eaa4-4459-b4da-00222be95df3	Duration: 1361.22 ms	Billed Duration: 1362 ms	Memory Size: 128 MB	Max Memory Used: 73 MB	Init Duration: 360.34 ms

Request ID
2636ef37-eaa4-4459-b4da-00222be95df3`
```

The documentation also states that this filter has been deprecated. 
`Lists all users in the identity store. Returns a paginated list of complete User objects. Filtering for a User by the UserName attribute is deprecated. Instead, use the GetUserId API action.
`
We suspect that the new IDC API has not been deployed in all regions yet, yet it has been deployed in the region I am using - us-west-1. This would explain why other users are not facing this issue yet.

**Comment by tawoyinfa** [comment]
@cemendes can you check that this fix #151 resolves the issue

**Comment by cemendes** [comment]
> @cemendes can you check that this fix #151 resolves the issue

Should i clean up my TEAM deployment and deploy again?

**Comment by tawoyinfa** [comment]
@cemendes you should be able to use the update script to pull and deploy the latest commit

**Comment by cemendes** [comment]
I messed my deployment directory while troubleshooting this issue and had to delete my TEAM instance and re-deploy it. I can confirm that the new fix allows me to log-in. 

Ps. The destroy script seems to be leaving sns and cloudtrail objects behind not allowing a new deployment to take place. I'll confirm and open a new issue.

**Comment by cemendes** [comment]
fix #151 allows me to log-in. closing this issue.
